impress-org/givewp
# Stripe Credit Card gateway now uses the Gateway API

Resolves #6181

Depends on #6179, #6189

## Description



This PR registers a new `stripe` gateway for handling credit card payments. This is the default gateway for Stripe.

The basic change is refactoring the gateway logic to use the new `GatewayPaymentData $paymentData` input and `GatewayCommand` return values.

The implementation of the gateway processing has also been refactored in order to support automated testing.

Other refactors include replacing nested `if/then/else` logic with early returns and custom exceptions (extending `PaymentGatewayException`). This co-locates the expectation of an error with the code block resulting in the error.

## Affects



The legacy implementation of the `stripe` gateway (`class-give-stripe-card.php`) has an early return which prevents the legacy process from running. This allows for the continued use of the legacy code for gateway specific settings.

## Testing Instructions



This replaces the Stripe Credit Card gateway implementation. Enable the Stripe Credit Card Gateway and complete a payment. There should be no change in behavior.

## Pre-review Checklist
